src/OrchardCore.Modules/OrchardCore.Redis/HealthChecks/RedisHealthCheck.cs

```diff
@@ -0,0 +1,29 @@
+using Microsoft.Extensions.Diagnostics.HealthChecks;
+using System.Threading.Tasks;
+using System.Threading;
+
+namespace OrchardCore.Redis.HealthChecks;
+
+public class RedisHealthCheck : IHealthCheck
+{
+    private readonly IRedisService _redisService;
+
+    public RedisHealthCheck(IRedisService redisService)
+    {
+        _redisService = redisService;
+    }
+
+    public async Task<HealthCheckResult> CheckHealthAsync(HealthCheckContext context, CancellationToken cancellationToken = default)
+    {
+        try
+        {
+            await _redisService.ConnectAsync();
```

Comment:
Then it will do nothing after being first connected, would need to think about it a little more.

Maybe something like this

    var time = await _redis.Database.PingAsync();
    if (time > TimeSpan.FromSeconds(30))
    {
        ...
    }

Reply:
That's why I didn't use `IRedisService` at the beginning coz I saw some catching after that first connection

Reply:
But still not good to create a new connection each time

We could check if the connection is still connected and so on

But in my above example we don't create a new connection but we are testing it

Hmm maybe good to also check if still connected

            if (_redis.Connection == null)
            {
                await _redis.ConnectAsync();
            }

            if (_redis.Connection.IsConnected)
            {
                var time = await _redis.Database.PingAsync();
                if (time > TimeSpan.FromSeconds(30))
                {
                    ...
                }
            }
            else
            {
                    ...
            }

Reply:
I need to test it then